py/redrock/external/desi.py

```diff
@@ -993,20 +993,20 @@ def rrdesi(options=None, comm=None):
 
         # Get the dictionary of wavelength grids (with keys as wavehashes)
         dwave = targets.wavegrids()
-        
+
         ncamera = len(list(dwave.keys())) # number of cameras for given instrument
         if args.archetypes_no_legendre or args.archetypes is None:
             if comm_rank == 0:
                 print('no archetypes or --archetypes-no-legendre; will turn off all the Legendre related arguments')
             archetype_legendre_prior = None
-            archetype_legendre_degree =0
+            args.archetype_legendre_degree =0
             archetype_legendre_percamera = False
```

Comment:
This change highlights that it is fragile to have some of the archetype options referenced through `args.archetype_...` and others without `args.`.  e.g. there is an option `--archetype-legendre-percamera`, but it appears that it is never used because the code only refers to `archetype_legendre_percamera`, not `args.archetype_legendre_percamera`.  Please review the usage of `--archetype-legendre-degree`, `--archetype-legendre-percamera`, `--archetype-legendre-prior`, and `--archetype-nnearest` to make sure they are all consistently used.  Consider using all of them like you changed here with `args.archetype_legendre_degree` without having a separate non-args version of the variable.